refactron/analysis/symbol_table.py

```
@@ -136,6 +194,61 @@ def build_for_project(self, project_root: Path) -> SymbolTable:
+
+    def _analyze_file(self, file_path: Path) -> None:
+        """Analyze a single file and populate symbols."""
+        file_path_str = file_path.resolve().as_posix()
+        try:
+            # We use astroid for better inference capabilities later
+            tree = self.inference_engine.parse_file(file_path_str)
+
+            # Walk the tree
+            self._visit_node(tree, file_path_str, "global")
+
+        except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)